### DISCUSSION TITLE
Integrate generated Mapstruct interface into Mapstructs annotation processing

### DISCUSSION THREAD

**Discussion by lostiniceland** [mapstruct/mapstruct#3346] [Q&A] [answered]
I've build a little annotation-processor which runs before Mapstructs processor which is generating Mapstruct interface-definitions for custom value-types (among other things).

Currently I have to add this interface to the `uses` field in `@Mapper` in order for Mapstruct to pick up those mapping rules. The problem is, that this interface is not available in a clean environment so I have no IDE support: I have to add the class knowing its FQN and when compiling everything works, because my processor runs before Mapstruct.

This is not ideal and I am looking for a way to hook Into Mapstruct directly from my processor, so that the user does not have to deal with it.

Is there any SPI mechanism for Mapstruct that I could use to "dynamically" register a mapper which is then automatically picked up once Mapstruct annotation-processor is running?

**Comment by filiphr** [comment, marked as answer]
@lostiniceland we don't have such functionality right now. However, I do think that https://github.com/mapstruct/mapstruct/issues/3197 is in the direction that you need

**Reply by lostiniceland** [reply]
Thanks for the issue-link. It is exactly what I need.
I guess we can live with having to "use" a type which is generated. I will probably just use a `@MapperConfig` so this smell is only in one location until #3197 is available.
